pkg/cache/sql/informer/listoption_indexer.go

```diff
@@ -417,12 +418,15 @@ func (l *ListOptionIndexer) buildORClauseFromFilters(orFilters OrFilter) (string
 			return "", nil, err
 		}
 
-		orWhereClause += fmt.Sprintf(`f."%s" %s ?`, columnName, opString)
+		orWhereClause += fmt.Sprintf(`f."%s" %s ? ESCAPE '\'`, columnName, opString)
 		format := strictMatchFmt
 		if filter.Partial {
 			format = matchFmt
 		}
-		params = append(params, fmt.Sprintf(format, filter.Match))
+		match := filter.Match
+		match = strings.ReplaceAll(match, `_`, `\_`)
+		match = strings.ReplaceAll(match, `%`, `\%`)
+		params = append(params, fmt.Sprintf(format, match))
```

Comment:
Should we support filtering fields that contain `\`?

Right now afaik it doesn't:

```
sqlite> create table T(t text);
sqlite> insert into T(t) values ('my\windows\path');
sqlite> select * from T where t LIKE 'my\windows\path' ESCAPE '\';
sqlite> select * from T where t LIKE 'my\\windows\\path' ESCAPE '\';
my\windows\path
```

Maybe we need. Would also be worth it to add a test case for this.

```suggestion
		match := filter.Match
		match = strings.ReplaceAll(match, `\`, `\\`)
		match = strings.ReplaceAll(match, `_`, `\_`)
		match = strings.ReplaceAll(match, `%`, `\%`)
		params = append(params, fmt.Sprintf(format, match))
```

Reply:
You don't need to use `ESCAPE` in your first query - then sqlite will return that row.
The backslash isn't special in sqlite, unlike the underscore and percent sign.

This PR aims to work around the special behavior of the LIKE operator when used with strings containing `_` and `%`.
Here are some docs describing it:
https://www.sqlite.org/lang_expr.html#the_like_glob_regexp_match_and_extract_operators

Reply:
I'm not sure what you mean by 

> You don't need to use ESCAPE in your first query - then sqlite will return that row.

In the context of this PR. The PR adds `ESCAPE` to the query unconditionally. So my example shows what the query will look like, meaning it will have where clauses with `ESCAPE`. Right now, this doesn't allow matching fields with `\` in them.

I added a quick test in there (I'm only showing the test but I added the rest of the code necessary to make that work):

```go
		{
			name: "match backslash",
			filters: orFiltersForFilters(informer.Filter{
				Field:   []string{`metadata`, `annotations[somekey]`},
				Match:   `my\backslash\path`,
				Op:      informer.Eq,
				Partial: false,
			}),
			wantNames: []string{"backslash"},
		},

```

and this fails with:

```
                        Error:          Not equal:                                                                                                                                                                                                                                                                     
                                        expected: 1                                                                                                                                                                                                                                                                    
                                        actual  : 0                                                                                                                                                                                                                                                                    
                        Test:           TestIntegrationSuite/TestSQLCacheFilters/match_backslash
```

because it can't find the configmap.

Reply:
Adding

```go
match = strings.ReplaceAll(match, `\`, `\\`)
```

makes that test succeed.

Reply:
Ah, I see, good point. Using ESCAPE indiscriminately in the query ignores the back slash itself when it's needed.
I think we could change that as you suggest.